### PR TITLE
Create a init test for cluster

### PR DESCRIPTION
**Pull request by thu-david**
### What changes are proposed in this pull request?

I have created a init test to check whether the workers has already run and connected to the leader.

### Why are the changes needed?

Because the version has updated to 3.0 but we still do not have the correct init test for the cluster.

### Does this PR introduce any user facing changes?

Yes, users may still not run test when they first run the cluster because i must communicate with leader to decide which args will be used.

### CODE REVIEW

**Comment by thu-david** [comment]
截屏2023-09-18 18 39 32
This is the first test of one worker.
截屏2023-09-18 18 41 22
This is the second test of two workers.
截屏2023-09-18 18 43 36
This is the second test of two workers and shutdown one worker.

**Comment by lucyge2022** [comment]
can i ask is this test to testing an existing cluster or mock a cluster to mock different worker live or dead situation?

File: dora/shell/src/main/java/alluxio/cli/CheckCluster.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    int totalWorkers = successfulWorkers + failedWorkers;
+    System.out.println("\nTotal workers: " + totalWorkers);
+    System.out.println(ANSI_GREEN + "Successful workers: " + successfulWorkers + ANSI_RESET);
+    System.out.println(ANSI_RED + "Failed workers: " + failedWorkers + ANSI_RESET);
+
+    if (!FAILED_WORKER.isEmpty()) {
+      System.out.println(ANSI_RED + "\nList of failed workers:" + ANSI_RESET);
+      for (BlockWorkerInfo failedWorker : FAILED_WORKER) {
+        System.out.println("Failed worker capacity and port:");
+        System.out.println(ANSI_RED + "Capacity:" + failedWorker.getCapacityBytes() + " Port:"
```
Comment (elega) [inline dora/shell/src/main/java/alluxio/cli/CheckCluster.java:206]: Let's print as the following

<hostname>:<port> Available space <available spce>/<capacity>

File: dora/shell/src/main/java/alluxio/cli/CheckCluster.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  }
+
+  private static void runTestWithCaching(FileSystem fs) throws Exception {
+    AlluxioURI testDir = initializeTestDirectory(fs);
+    HashMap<BlockWorkerInfo, AlluxioURI> workerMap = assignToWorkers(fs, testDir, false);
+    verifyFileAssignments(fs, workerMap, false);
+    for (Map.Entry<BlockWorkerInfo, AlluxioURI> entry : workerMap.entrySet()) {
+      verifyFileContent(fs, entry.getValue());
+    }
+    System.out.println("cache through test");
```
Comment (elega) [inline dora/shell/src/main/java/alluxio/cli/CheckCluster.java:219]: "Test with CACHE_THROUGH write type"

File: dora/shell/src/main/java/alluxio/cli/CheckCluster.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    printInfo();
+  }
+
+  private static void runTestWithoutCaching(FileSystem fs, FileSystemContext fsContext)
+      throws Exception {
+    System.out.println("through test");
```
Comment (elega) [inline dora/shell/src/main/java/alluxio/cli/CheckCluster.java:225]: "Test with THROUGH write type"

File: dora/shell/src/main/java/alluxio/cli/CheckCluster.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Example to show the basic operations of Alluxio.
```
Comment (elega) [inline dora/shell/src/main/java/alluxio/cli/CheckCluster.java:38]: Update the comment here and explain
1. What does this command do
2. How were you able to hit and cover all workers in this test

File: dora/shell/src/main/java/alluxio/cli/CheckCluster.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  }
+
+  private static HashMap<BlockWorkerInfo, AlluxioURI> assignToWorkers(FileSystem fs,
+                                                                      AlluxioURI testDir,
+                                                                      boolean throughOnly)
+      throws Exception {
+    HashMap<BlockWorkerInfo, AlluxioURI> workerMap = new HashMap<>();
+    sPolicy = WorkerLocationPolicy.Factory.create(Configuration.global());
+
+    for (int i = 0; i < NUM_FILES && workerMap.size() < sWorkerInfoList.size(); i++) {
```
Comment (elega) [inline dora/shell/src/main/java/alluxio/cli/CheckCluster.java:121]: 1. what if not all workers are covered after NUM_FILES-th iterations? 
2. can you make this one configurable and recommend the value for different numbers of workers? (assume data are evenly distributed across the hash ring)

File: dora/shell/src/main/java/alluxio/cli/CheckCluster.java [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        runTestWithoutCaching(fs, fsContext);
+      } catch (Exception e) {
+        System.out.println("Exception occurred during through test.");
+        handleException(e);
+      }
+    } catch (Exception e) {
+      System.out.println(
+          "Exception occurred during FileSystemContext creation or worker retrieval.");
+      handleException(e);
+    }
```
Comment (elega) [inline dora/shell/src/main/java/alluxio/cli/CheckCluster.java:88]: add a finally block to clean up the directory?

**Review by elega — APPROVED** [review]

**Comment by elega** [comment]
alluxio-bot, merget this please.

**Comment by elega** [comment]
alluxio-bot, merge this please.

**Pull request merged**